src/domain_bridge.cpp

```diff
@@ -104,3 +152,69 @@ int main(int argc, char ** argv)
 
   return 0;
 }
+
+const char *
+get_domain_profile_prefix()
+{
+  const char * qos_profile_library_name = nullptr;
+  const char * error = rcutils_get_env(
+    "DOMAIN_BRIDGE_CONNEXT_DDS_QOS_PROFILE_PREFIX", &qos_profile_library_name);
+  if (error) {
+    throw std::runtime_error(std::string("rcutils_get_env() failed: ") + error);
+  }
+  if (qos_profile_library_name && 0 == strcmp("", qos_profile_library_name)) {
+    return nullptr;
+  }
+  return qos_profile_library_name;
+}
+
+bool
+set_default_qos_library(DDS_DomainParticipantFactory * dpf)
+{
+  const char * qos_profile_library_name = nullptr;
+  const char * error = rcutils_get_env(
+    "DOMAIN_BRIDGE_CONNEXT_DDS_QOS_LIBRARY_NAME", &qos_profile_library_name);
+  if (error) {
+    throw std::runtime_error(std::string("rcutils_get_env() failed: ") + error);
+  }
+  if (qos_profile_library_name && 0 == strcmp("", qos_profile_library_name)) {
+    qos_profile_library_name = nullptr;
+  }
+
+  DDS_StringSeq qos_libraries;
+  rcpputils::scope_exit([qos_libraries = &qos_libraries]() {
+    DDS_StringSeq_finalize(qos_libraries);
+  });
+  if (!qos_profile_library_name) {
+    // environment variable is empty
+    if (DDS_RETCODE_OK != DDS_DomainParticipantFactory_get_qos_profile_libraries(dpf, &qos_libraries)) {
+      throw std::runtime_error("failed to get qos profile libraries");
+    }
+    // If only one non-builtin qos profile library was loaded, use that one.
+    if (DDS_StringSeq_get_length(&qos_libraries) > 3) {
+      return false;
+    }
+    for (int i = 0; i < DDS_StringSeq_get_length(&qos_libraries); i++) {
+      const char * item = DDS_StringSeq_get(&qos_libraries, i);
+      if (
+        strcmp("BuiltinQosLib", item) != 0 &&
+        strcmp("BuiltinQosLibExp", item) != 0)
+      {
+        if (!qos_profile_library_name) {
+          qos_profile_library_name = item;
+        } else {
+          // More than one QoS profile library was loaded, do not set
```

Comment:
would be helpful to print a warning here

Reply:
Done in https://github.com/ros2/domain_bridge/pull/25/commits/33413665a0760b396cba9c5510ea594af8f4c2e4